ouroboros-network/src/Ouroboros/Network/Diffusion/Configuration.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE DeriveGeneric  #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Ouroboros.Network.Diffusion.Configuration where
+
+import Control.Monad.Class.MonadTime.SI (DiffTime)
+import Data.Monoid (Last (..))
+import GHC.Generics (Generic)
+
+import System.Random (randomRIO)
+
+import Ouroboros.Network.BlockFetch.Types (BlockFetchConfiguration (..))
+import Ouroboros.Network.NodeToNode (MiniProtocolParameters (..))
+import Ouroboros.Network.PeerSelection.PeerMetric
+           (PeerMetricsConfiguration (..))
+import Ouroboros.Network.PeerSelection.PeerSharing (PeerSharing (..))
+import Ouroboros.Network.Protocol.ChainSync.Codec (ChainSyncTimeout (..))
+import Ouroboros.Network.Protocol.Limits (shortWait)
+import Ouroboros.Network.Server.RateLimiting (AcceptedConnectionsLimit (..))
+
+
+data NetworkP2PMode = EnabledP2PMode | DisabledP2PMode
+  deriving (Eq, Show, Generic)
+
+-- | Basic configuration parameters of diffusion algorithm taken from cardano- node moving here where they belong
+data DiffusionConfiguration =
+  DiffusionConfiguration {
+    dcProtocolIdleTimeout                    :: !(Last DiffTime),
+    dcTimeWaitTimeout                        :: !(Last DiffTime),
+
+    -- AcceptedConnectionsLimit
+    dcAcceptedConnectionsLimit               :: !(Last AcceptedConnectionsLimit),
+
+    -- P2P governor targets
+    dcTargetCountOfRootPeers                 :: !(Last Int),
+    dcTargetCountOfKnownPeers                :: !(Last Int),
+    dcTargetCountOfEstablishedPeers          :: !(Last Int),
+    dcTargetCountOfActivePeers               :: !(Last Int),
+    dcTargetCountOfKnownBigLedgerPeers       :: !(Last Int),
+    dcTargetCountOfEstablishedBigLedgerPeers :: !(Last Int),
+    dcTargetCountOfActiveBigLedgerPeers      :: !(Last Int),
+
+    -- Enable experimental P2P mode
+    dcEnableP2P                              :: !(Last NetworkP2PMode),
+
+    -- Peer Sharing
+    dcPeerSharing                            :: !(Last PeerSharing) }
+
+mkDefaultNetworkConfig :: DiffusionConfiguration
+mkDefaultNetworkConfig =
+  DiffusionConfiguration {
+    dcProtocolIdleTimeout = Last $ Just 5,
+    dcTimeWaitTimeout = Last $ Just 60,
+    dcAcceptedConnectionsLimit =
+      Last $
+      Just $
+      AcceptedConnectionsLimit {
+        acceptedConnectionsHardLimit = 512,
+        acceptedConnectionsSoftLimit = 384,
+        acceptedConnectionsDelay     = 5 },
+    dcTargetCountOfRootPeers        = Last $ Just 85,
+    dcTargetCountOfKnownPeers       = Last $ Just 85,
+    dcTargetCountOfEstablishedPeers = Last $ Just 40,
+    dcTargetCountOfActivePeers      = Last $ Just 15,
+    dcTargetCountOfKnownBigLedgerPeers       = Last $ Just 15,
+    dcTargetCountOfEstablishedBigLedgerPeers = Last $ Just 10,
+    dcTargetCountOfActiveBigLedgerPeers      = Last $ Just 5,
+    dcEnableP2P   = Last $ Just DisabledP2PMode,
+    dcPeerSharing = Last $ Just PeerSharingDisabled }
```

Comment:
I don't think we should bring all this stuff from `cardano-node`.

Some of them are already defined in `ouroboros-network`, e.g.
* [connection manager][https://github.com/intersectmbo/ouroboros-network/blob/master/ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs#L446-L458] - they probably need to be defined in `ouroboros-network-framework`, then we just need to re-export them.
* We already have `PeerSelectionTargets` record, let's just define `defaultPeerSelectionTargets` in this module, and in the `cardano-node` we can use it in `mkDefaultNetworkConfig`.
* I found one more constant in [Handshake][https://github.com/intersectmbo/ouroboros-network/blob/f83f71ad674721605efa0f31cf2144cc6dd5dd4d/ouroboros-network-framework/src/Ouroboros/Network/Protocol/Handshake.hs#L190-L191].

Reply:
Note that the advantage of re-using `PeerSelectionTarget`, is that whenever we modify it we will be forced to add a default value. @bolt12 changes to bootstrap are of this kind.